contracts/swap/src/swap.rs

```diff
@@ -131,57 +135,73 @@ pub fn handle_atomic_order_reply(
         err: err.to_string(),
     })?;
 
-    // unwrap results into trade_data
     let trade_data = match order_response.results.into_option() {
         Some(trade_data) => Ok(trade_data),
         None => Err(ContractError::CustomError {
             val: "No trade data in order response".to_string(),
         }),
     }?;
-    let quantity = FPDecimal::from_str(&trade_data.quantity)? / dec_scale_factor; // need to remove protobuf scale factor to get real values
-    let avg_price = FPDecimal::from_str(&trade_data.price)? / dec_scale_factor;
+
+    // need to remove protobuf scale factor to get real values
+    let average_price = FPDecimal::from_str(&trade_data.price)? / dec_scale_factor;
+    let quantity = FPDecimal::from_str(&trade_data.quantity)? / dec_scale_factor;
```

Comment:
maybe we could use `proto_to_dec` function?

Reply:
damn its not yet in cw-injective though